main.tf

```diff
@@ -2,7 +2,79 @@ locals {
   cluster_name = "aks-${var.names.resource_group_type}-${var.names.product_name}-${var.names.environment}-${var.names.location}"
 }
 
+resource "azurerm_role_assignment" "subnet_network_contributor" {
+  count                = (var.use_service_principal ? (var.create_default_node_pool_subnet ? 0 : 1) : 0)
+  scope                = var.default_node_pool_subnet.id
+  role_definition_name = "Network Contributor"
+  principal_id         = data.azuread_service_principal.aks[0].object_id
+}
+
+resource "azurerm_network_security_rule" "outbound_allow_azure_cloud_udp_1194" {
+  count                       = (var.create_default_node_pool_subnet ? 0 : 1)
+  name                        = "AKS_Control_Plane_UDP_1194"
+  priority                    = var.nsg_rule_priority_start
+  direction                   = "Outbound"
+  access                      = "Allow"
+  protocol                    = "UDP"
+  source_port_range           = "*"
+  destination_port_range      = "1194"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "AzureCloud"
+  resource_group_name         = var.default_node_pool_subnet.resource_group_name
+  network_security_group_name = var.default_node_pool_subnet.security_group_name
+}
+
+resource "azurerm_network_security_rule" "outbound_allow_azure_cloud_tcp_9000" {
+  count                       = (var.create_default_node_pool_subnet ? 0 : 1)
+  name                        = "AKS_Control_Plane_TCP_9000"
+  priority                    = (var.nsg_rule_priority_start + 1)
+  direction                   = "Outbound"
+  access                      = "Allow"
+  protocol                    = "TCP"
+  source_port_range           = "*"
+  destination_port_range      = "9000"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "AzureCloud"
+  resource_group_name         = var.default_node_pool_subnet.resource_group_name
+  network_security_group_name = var.default_node_pool_subnet.security_group_name
+}
+
+resource "azurerm_network_security_rule" "outbound_allow_ntp" {
+  count                       = (var.create_default_node_pool_subnet ? 0 : 1)
+  name                        = "AKS_NTP"
+  priority                    = (var.nsg_rule_priority_start + 2)
+  direction                   = "Outbound"
+  access                      = "Allow"
+  protocol                    = "UDP"
+  source_port_range           = "*"
+  destination_port_range      = "123"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  resource_group_name         = var.default_node_pool_subnet.resource_group_name
+  network_security_group_name = var.default_node_pool_subnet.security_group_name
+}
+
+resource "azurerm_network_security_rule" "outbound_allow_all_tcp_443" {
+  count                       = (var.create_default_node_pool_subnet ? 0 : 1)
+  name                        = "AKS_SSL"
+  priority                    = (var.nsg_rule_priority_start + 3)
```

Comment:
Would it be a good idea here, instead of nsg_rule_priority_start +1, +2, +3, to do +5, +10, +15 in case some day something must be inserted between existing rules? That's usually the best practice when setting rule priorities. It's unlikely to become a problem in this case but you never know...

Reply:
that's a very good idea, will add.